miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/ContactRoleRestControllerST.java

```diff
@@ -74,15 +74,15 @@ public void testBulkCreateFail() throws Exception {
     // contact roles can only be created by an admin user, so this test expected failure due to
     // insufficient permissions
 
-    testBulkCreateAsyncUnauthorized(CONTROLLER_BASE, controllerClass, makeCreateDtos());
+    testBulkCreateAsyncUnauthorized(CONTROLLER_BASE, entityClass, makeCreateDtos());
   }
 
   @Test
   @WithMockUser(username = "admin", password = "admin", roles = {"INTERNAL", "ADMIN"})
   public void testBulkUpdateAsync() throws Exception {
     // contact roles can only be updated by an admin user
-    ContactRole con2 = currentSession().get(controllerClass, 2);
-    ContactRole con3 = currentSession().get(controllerClass, 3);
+    ContactRole con2 = currentSession().get(entityClass, 2);
+    ContactRole con3 = currentSession().get(entityClass, 3);
     con2.setName("con2");
     con3.setName("con3");
```

Comment:
Didn't notice this in other PRs, but you should be modifying the fields on the DTOs - not on the entity classes. This is very important. As is, you're updating the entities within the session, so these changes could be persisted before (and without) any interaction with the controller.

Please check other merged tests and correct if necessary.